src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -109,6 +112,15 @@ void EventPipe::FinishInitialize()
 {
     STANDARD_VM_CONTRACT;
 
+    s_CanStartThreads = true;
+
+    while (s_rgDeferredEventPipeSessionIds.Size() > 0)
+    {
+        EventPipeSessionID id = s_rgDeferredEventPipeSessionIds.Pop();
```

Comment:
Yup, I caught that this morning.  I think that's why the Windows checked runs were failing.  (That and a contract violation).

Reply:
Sessions cannot be stopped until the runtime has been resumed now.

Reply:
> Sessions cannot be stopped until the runtime has been resumed now.

That feels like an odd restriction that there is a special time period during which it is legal to start sessions but not to stop them. Rather than adding complexity to the user visible behavior how hard would it be to remove the id out of this list or rely on the IsSessionIdInCollection(id) call?

Reply:
This list is used to lazily start streaming events once the runtime is capable of creating threads.  The session itself has already been created and enabled inside EventPipe which means we've modified some internal state of EventPipe.  Disabling the session at this point will involve removing any providers from the session as well.  Right now, all that logic is inside `EventPipe::Disable` which also attempts to do things like Rundown which I'm not sure will be able to execute at this point.  I'll see if it is trivial to add a modified disable path that closes this loop without too much overhead, e.g., always skips Rundown and unregisters any enabled providers from the `RunWithCallbackDeferred` queue.

Reply:
I also think it would be OK if the disable call wasn't able to finish until after startup resumed (as long as the blocked disable call doesn't prevent other diagnostic server commands from being handled)

Reply:
Just pushed some changes that allow you stop sessions while in the paused state.